server/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java

```diff
@@ -665,4 +664,22 @@ public void test_column_pruning_is_applied_to_subqueries() throws Exception {
                 "      └ Collect[doc.t1 | [a] | (x > 10)]"
             );
     }
+
+    @Test
+    public void test_rename_gets_scalar_outputs_from_collect() {
+        var plan = sqlExecutor.logicalPlan("""
+          SELECT
+            TRY_CAST(x AS BIGINT) AS val
+          FROM t1 as t
+          WHERE x < 123
+          ORDER BY val ASC
+        """);
+        assertThat(plan)
+            .hasOperators(
+                "Eval[try_cast(x AS bigint) AS val]",
+                "  └ Rename[try_cast(x AS bigint) AS val, try_cast(x AS bigint)] AS t",
+                "    └ OrderBy[try_cast(x AS bigint) ASC]",
+                "      └ Collect[doc.t1 | [try_cast(x AS bigint) AS val, try_cast(x AS bigint)] | (x < 123)]"
+            );
+    }
```

Comment:
I also tests with integration test locally, it works  with this change (and doesn't on master).

Also tested query from https://github.com/crate-workbench/cratedb-toolkit/actions/runs/9377632509/job/25819513539, also works